src/Logging.chpl

```diff
@@ -0,0 +1,88 @@
+module Logging {
+    use Set;
+    use IO;
+    use DateTime;
+
+    /*
+     * The LogLevel enum is used to provide a strongly-typed means of
+     * configuring the logging level of a Logger object
+     */
+    enum LogLevel {DEBUG,INFO,WARN,ERROR,CRITICAL};
+    
+    /*
+     * The Logger class provides structured log messages at various levels
+     * of logging sensitivity analogous to other languages such as Python.
+     */
+    class Logger {
```

Comment:
I think this could be a record instead of a class. That avoids a heap allocation, but shouldn't have a major perf impact.